core/data/types.go

```diff
@@ -18,6 +18,12 @@ const (
 	ARRAY
 	PARAMS
 	COMPLEX_OBJECT
+
```

Comment:
These don't seem like types, ref, arrayref, function or expression. An 'array' is a type, params, etc. These seem more like mapper specific details?